### ISSUE TITLE
Some people have avatars in the member list, read receipt, and 1:1 chats but not when they send a message

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#7418]
For example:
![image](https://user-images.githubusercontent.com/1190097/46237877-903e6200-c344-11e8-8776-c7691a1fc95e.png)

The sender and RR are the same person in this screenshot - surely they should have an avatar on the tile?

**Comment by bwindels** [comment]
This seems to be #7417 . If you've got any hints towards reproducing this I would be very grateful @turt2live

**Comment by bwindels** [comment]
@turt2live can you confirm the blurred out name was indeed the user's mxid?

**Comment by turt2live** [comment]
The blurred out name was the user's display name, not mxid

**Comment by bwindels** [comment]
Ok, then it's not a missing sentinel bug, but probably ending up using an earlier membership event that didn't have an avatar yet. @turt2live I guess you don't remember if that user had set an avatar earlier in that room?

**Comment by turt2live** [comment]
This happened to me again, and it looks like it's only partially Riot's fault. The user did have an avatar set, and the media repo was giving a periodic 500 error for that particular size of avatar. The request would randomly fail due to a bug in the media repo.

Aside from me fixing my media repo, Riot should probably avoid caching the 500 error if it can and retry periodically for the avatar. As far as I could tell, the avatar ended up deep within Chrome's cache and is likely to appear broken for a day or so.

**Comment by bwindels** [comment]
That's very useful info, thanks. I suggest de-prioritizing this, wdyt @lampholder?

**Comment by lampholder** [comment]
I agree - esp. if we've fixed the media repo such that this is much less likely to occur.
